test/ranges.jl

```diff
@@ -1597,8 +1624,6 @@ end
             end
         end
     end
-    # require a keyword arg
-    @test_throws ArgumentError range(1, 100)
```

Comment:
This test arguably contradicts the docs. It is documented that step will default to `1` so this specifies a unique range.

Reply:
On top of that on master we have the following interesting behavior:
```julia
julia> range(1,stop=100)
1:100

julia> range(1,100)
ERROR: ArgumentError: At least one of `length` or `step` must be specified
```

Reply:
Ref. https://github.com/JuliaLang/julia/pull/28708#issuecomment-420034562 and the discussion following it.

Reply:
Ah, thanks a lot. While I don't favor the design I understand the reasoning and will disallow `range(1, 100)`.